### PR TITLE
Invariants::after takes the operation input simplex.

### PR DESCRIPTION
**Pull request by daniel-zint**

### CODE REVIEW

**Comment by mtao** [comment]
should it take the input simplex or should it take the input equivalent set of simplices? we're mixing semantics atm i think

**Comment by daniel-zint** [comment]
> should it take the input simplex or should it take the input equivalent set of simplices? we're mixing semantics atm i think

I was thinking about that too. The thing is, we cannot get the equivalent without knowing the operation. Let's discuss that tomorrow.

File: src/wmtk/operations/tri_mesh/EdgeSwapBase.cpp [hunk 1]
```diff
@@ -103,9 +103,12 @@ bool EdgeSwapBase::execute()
...
 {
-    std::vector<Simplex> s;
-    s.emplace_back(simplex::Simplex::edge(m_output_tuple));
-    return s;
+    return {simplex::Simplex::edge(m_output_tuple)};
+}
+
+std::vector<Simplex> EdgeSwapBase::unmodified_primitives() const
+{
+    return {input_simplex()};
```
Comment (mtao) [inline src/wmtk/operations/tri_mesh/EdgeSwapBase.cpp:111]: weren't you the one who had a compiler complain about this return notation? i've been avoiding this for you - but i guess i can do this again ;-) 
(i think this was particularly with std::array your compiler was not happy)

File: src/wmtk/optimization/LineSearch.cpp [hunk 2]
```diff
@@ -28,7 +28,9 @@ bool LineSearch::check_state() const
                                                // vertex here but I didn't know for sure.
     bool after_pass = true;
     for (const PrimitiveType pt : wmtk::utils::primitive_below(top_type)) {
-        after_pass &= m_invariants.after(pt, modified_simplices(pt));
+        after_pass &= m_invariants.after(
```
Comment (mtao) [inline src/wmtk/optimization/LineSearch.cpp:31]: don't worry about this guess, it's being deleted anyway ;-)

**Review by mtao — APPROVED** [review]
seems in good state to me

File: src/wmtk/operations/tri_mesh/EdgeSwapBase.cpp (hunk 1 above)
Comment (daniel-zint) [reply src/wmtk/operations/tri_mesh/EdgeSwapBase.cpp:111]: No my compiler never complained about that stuff. It was only that I didn't know that for `std::array` you should write `{{...}}` but it never was an issue.

**Pull request merged**